modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/StemmerOverrideTokenFilterFactory.java

```diff
@@ -57,19 +57,23 @@ public TokenStream create(TokenStream tokenStream) {
 
     static void parseRules(List<String> rules, StemmerOverrideFilter.Builder builder, String mappingSep) {
         for (String rule : rules) {
-            String key, override;
-            List<String> mapping = Strings.splitSmart(rule, mappingSep, false);
-            if (mapping.size() == 2) {
-                key = mapping.get(0).trim();
-                override = mapping.get(1).trim();
-            } else {
+            List<String> sides = Strings.splitSmart(rule, mappingSep, false);
+            if (sides.size() != 2) {
                 throw new RuntimeException("Invalid Keyword override Rule:" + rule);
             }
 
-            if (key.isEmpty() || override.isEmpty()) {
+            List<String> keys = Strings.splitSmart(sides.get(0), ",", false);
```

Comment:
Ugh... Honestly, I didn't even look at `splitSmart` does and I simply used that because it was used to split rules on `=>`. Now I took a look (although that helper itself does not have any test) and I'm confused.
Unlike `String.split` `splitSmart` does not return empty tokens. So the result of splitting:
```
=>one=>two=>=>
```
is `[one, two]` which is considered a valid rule. IMO it should not.

I believe a regular `String.split` should be used. If we want to support `"a\\,b"` tokens then we should split on `""(?<!\\\\),"` regexp and then we should unescape each token before adding it - this is something that I would normally use [`StringEscapeUtils.unescapeJava`](https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/StringEscapeUtils.html#unescapeJava-java.lang.String-) from apache commons-text, but you don't seem to use it.

Reply:
> =>one=>two=>=>

Good point, maybe we should use simple String.split for the LHS/RHS separation and reject invalid patterns indeed.
For the LHS split on `,` I'm find both ways. If "splitSmart" works for this and supports escaping we can still consider using that, but simple String splitting on `,` is also fine if other solutions add too much complexity. I don't think we really need to support commas in LHS entries if its too complicated.